### PR TITLE
Check if copysets on chunkservers consistent with mds when check copyset health.

### PR DESCRIPTION
**Pull request by bai-charisu**
…sets health

Change-Id: I8fdb353323595b583dd2a31f5b6b397b8539bade

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: src/tools/copyset_check_core.cpp [hunk 1]
```diff
@@ -416,6 +422,38 @@ int CopysetCheckCore::CheckCopysetsInCluster() {
...
+    CopysetStatistics statistics = GetCopysetStatistics();
+    if (copysetsInMds.size() != copysets_[kTotal].size()) {
+        std::cout << "Copyset numbers in chunkservers not consistent"
+                     " with mds, please check! copysets on chunkserver: "
+                     << copysets_[kTotal].size() << ", copysets in mds: "
+                     << copysetsInMds.size() << std::endl;
+        return -1;
+    }
+    if (copysetsInMds.empty()) return copysets_[kTotal].empty();
+    if (copysets_[kTotal].empty()) return copysetsInMds.empty();
```
Comment (ilixiaocui) [inline src/tools/copyset_check_core.cpp:441]: 到这里说明两个的copyset数量相等，一个if就可以了？

File: src/tools/copyset_check_core.cpp [hunk 2]
```diff
@@ -416,6 +422,38 @@ int CopysetCheckCore::CheckCopysetsInCluster() {
...
+    std::set<std::string> copysetsInMdsGid;
+    for (const auto& copyset : copysetsInMds) {
+        std::string gId = ToGroupId(copyset.logicalpoolid(),
+                                    copyset.copysetid());
+        copysetsInMdsGid.insert(gId);
+    }
+    if (*(copysetsInMdsGid.begin()) != *(copysets_[kTotal].begin())
```
Comment (ilixiaocui) [inline src/tools/copyset_check_core.cpp:448]: std::set_difference(v1.begin(), v1.end(), v2.begin(), v2.end(),
                        std::inserter(diff, diff.begin()));
这个可以比较两者的差异化

File: src/tools/copyset_check_core.cpp [hunk 3]
```diff
@@ -416,6 +422,38 @@ int CopysetCheckCore::CheckCopysetsInCluster() {
...
+            || *(--copysetsInMdsGid.end()) != *(--copysets_[kTotal].end())) {
```
Comment (xu-chaojie) [inline src/tools/copyset_check_core.cpp:449]: 这是只比较了头尾两个成员？ 中间的不比较？

File: src/tools/copyset_check_core.cpp (hunk 1 above)
Comment (bai-charisu) [reply src/tools/copyset_check_core.cpp]: 改成求差集后去掉了

File: src/tools/copyset_check_core.cpp (hunk 2 above)
Comment (bai-charisu) [reply src/tools/copyset_check_core.cpp]: done

File: src/tools/copyset_check_core.cpp (hunk 3 above)
Comment (bai-charisu) [reply src/tools/copyset_check_core.cpp]: 改成求差集了

**Review by ilixiaocui — APPROVED** [review]

**Review by xu-chaojie — APPROVED** [review]

**Pull request merged**